File: remote-health-frontend/src/types/message.ts
export type { UserRole } from './auth'; // Re-export UserRole to make it available
import { UserRole as LocalUserRole } from './auth'; // Import for local use

export interface Message {
  id: string; // Or number, depending on backend
  senderId: number;
  receiverId: number;
  senderRole?: LocalUserRole; // Optional: useful for UI differentiation
  receiverRole?: LocalUserRole; // Optional
  content: string;
  timestamp: string; // ISO date string
  isRead: boolean;
}

// Represents a user with whom the current user can chat
export interface ChatPartner {
  id: number; // User ID of the chat partner
  name: string;
  role: LocalUserRole;
  lastMessage?: string; // Optional: for conversation list preview
  lastMessageTimestamp?: string; // Optional
  unreadCount?: number; // Optional
  profilePictureUrl?: string; // Added for profile pictures
}

// Represents a full conversation with a partner
export interface Conversation {
  partner: ChatPartner;
  messages: Message[];
} 

File: RemoteHealthBackend-main/app/schemas/message.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from app.schemas.user import User  # Assuming User schema can be used or adapted
from app.models.user_model import UserRole # For UserRole enum

class MessageBase(BaseModel):
    content: str

class MessageCreate(MessageBase):
    receiver_id: int

class Message(MessageBase):
    id: int
    sender_id: int
    receiver_id: int
    timestamp: datetime
    is_read: bool
    sender_role: Optional[UserRole] = None
    receiver_role: Optional[UserRole] = None

    class Config:
        from_attributes = True

class ChatPartner(BaseModel):
    id: int
    name: str # This will be combined first_name and last_name
    role: UserRole
    last_message: Optional[str] = None
    last_message_timestamp: Optional[datetime] = None
    unread_count: Optional[int] = 0

    class Config:
        from_attributes = True

class MessageInDB(Message):
    pass 